pytorch_lightning/callbacks/batch_size_finder.py

```diff
@@ -0,0 +1,423 @@
+        """
+        supported_modes = ("power", "binsearch")
+        mode = mode.lower()
+        if mode not in supported_modes:
+            raise MisconfigurationException(f"`mode` should be one of {supported_modes}")
+
+        self.mode = mode
+        self.steps_per_trial = steps_per_trial
+        self.init_val = init_val
+        self.max_trials = max_trials
+        self.batch_arg_name = batch_arg_name
+        self.optimal_batch_size = init_val
+
+        self._early_exit = False
+
+        from pytorch_lightning.loggers.base import LightningLoggerBase
+
+        class _BatchSizeFinderDumpedParams(TypedDict):
+            callbacks: List[Callback]
+            logger: Optional[LightningLoggerBase]
+            max_steps: int
+            global_step: Optional[None]
+            limit_val_batches: Union[int, float]
+            limit_eval_batches: Union[int, float]
+
+        self._dumped_params: _BatchSizeFinderDumpedParams = {}
+
+    def scale_batch_size(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule") -> None:
+        if trainer.fast_dev_run:
+            rank_zero_warn("Skipping batch size scaler since `fast_dev_run` is enabled.")
+            return
+
+        if not lightning_hasattr(pl_module, self.batch_arg_name):
+            raise MisconfigurationException(
+                f"Field {self.batch_arg_name} not found in both `model` and `model.hparams`"
+            )
+
+        if (
+            hasattr(pl_module, self.batch_arg_name)
+            and hasattr(pl_module, "hparams")
+            and self.batch_arg_name in pl_module.hparams
+        ):
+            rank_zero_warn(
+                f"Field `model.{self.batch_arg_name}` and `model.hparams.{self.batch_arg_name}` are mutually exclusive!"
+                f" `model.{self.batch_arg_name}` will be used as the initial batch size for scaling."
+                " If this is not the intended behavior, please remove either one."
+            )
+
+        # TODO: check if this can be enabled (#4040)
+        if not trainer._data_connector._train_dataloader_source.is_module():
+            raise MisconfigurationException(
+                "The batch scaling feature cannot be used with dataloaders passed directly to `.fit()`."
+                " Please disable the feature or incorporate the dataloader into the model."
+            )
+
+        # Arguments we adjust during the batch size finder, save for restoring
+        self._dump_params(trainer)
+
+        # Set to values that are required by the algorithm
+        self._reset_params(trainer)
+
+        # Save initial model, that is loaded after batch size is found
+        save_path = os.path.join(trainer.default_root_dir, f".scale_batch_size_temp_model_{uuid.uuid4()}.ckpt")
+        trainer.save_checkpoint(save_path)
+
+        if trainer.progress_bar_callback:
+            trainer.progress_bar_callback.disable()
+
+        new_size, _ = self._adjust_batch_size(trainer, value=self.init_val)
```

Comment:
```suggestion
        optimal_batch_size, _ = self._adjust_batch_size(trainer, value=self.init_val)
```

For consistency.

Reply:
well `new_size` might not be the `optimal_batch_size`. More runs can happen after this one.

Reply:
Name it `new_batch_size` instead.